step-templates/aws-configure-lambda-api-gateway-integration.json

```diff
@@ -0,0 +1,121 @@
eKey.Replace(\"/\", \"-\"))\"\n}\nelse\n{\n\t$routeKeyToUse = \"\"\n    $statementIdToUse = \"$ApiGatewayName\"\n}\n$sourceArn = \"arn:aws:execute-api:$($apiRegion):$($accountInfo.Account):$($apiId)/*/*$routeKeyToUse\"\n\nWrite-Host \"Source ARN: $sourceArn\"\n$hasExistingPolicy = $false\n$deleteExistingPolicy = $false\n\ntry\n{\n\tWrite-Host \"Getting existing policies\"\n\t$existingPolicy = aws lambda get-policy --function-name \"$apiLambdaIntegrationArn\"\n    \n    if ($LASTEXITCODE -eq 255 -or $LASTEXITCODE -eq 254)\n    {\n    \tWrite-Host \"Last exit code was $LASTEXITCODE the policy does not exist\"\n    \t$hasExistingPolicy = $false\n    }\n    else\n    {\n    \tWrite-Host \"The policy exists\"\n    \t$hasExistingPolicy = $true\n    }\n    \n    $existingPolicy = ($existingPolicy | ConvertFrom-JSON)\n    Write-Host $existingPolicy\n    \n    $policyObject = ($existingPolicy.Policy | ConvertFrom-JSON)\n    \n\t$statementList = @($policyObject.Statement)\n    Write-Host \"Statement List $statementList\"\n    \n    foreach ($existingStatement in $statementList)\n    {\n    \tWrite-Host $existingStatement\n    \tWrite-Host \"Comparing $($existingStatement.Sid) with $statementIdToUse and $($existingStatement.Condition.ArnLike.'AWS:SourceArn') with $sourceArn\"\n    \tif ($existingStatement.Sid -eq \"$statementIdToUse\" -and $existingStatement.Condition.ArnLike.'AWS:SourceArn' -ne \"$sourceArn\")\n        {\n        \tWrite-Host \"The policy exists but it is not pointing to the write source arn, recreating it.\"\n        \t$deleteExistingPolicy = $true\n        }\n    }\n}\ncatch\n{\n\tWrite-Host \"Error pulling back the policies, this typically means the policy does not exist\"\n\t$hasExistingPolicy = $false\n}\n\nif ($hasExistingPolicy -eq $true -and $deleteExistingPolicy -eq $true)\n{\n\tWrite-Highlight \"Removing the existing policy $statementIdToUse\"\n    aws lambda remove-permission --function-name \"$apiLambdaIntegrationArn\" --statement-id \"$statementIdToUse\"\n}\n\nif ($hasExistingPolicy -eq $false -or $deleteExistingPolicy -eq $true)\n{\n\tWrite-Highlight \"Adding the policy $statementIdToUse\"\n\taws lambda add-permission --function-name \"$apiLambdaIntegrationArn\" --statement-id \"$statementIdToUse\" --action \"lambda:InvokeFunction\" --principal \"apigateway.amazonaws.com\" --qualifier \"$ApiLambdaAlias\" --source-arn \"$sourceArn\"\n}\n\nWrite-Highlight \"Setting the output variable 'Octopus.Action[$($stepName)].Output.ApiGatewayEndPoint' to $($apiGatewayToUpdate.ApiEndpoint)\"\nSet-OctopusVariable -name \"ApiGatewayEndPoint\" -value \"$($apiGatewayToUpdate.ApiEndpoint)\"\n\nWrite-Highlight \"Setting the output variable 'Octopus.Action[$($stepName)].Output.ApiGatewayId' to $apiId\"\nSet-OctopusVariable -name \"ApiGatewayId\" -value \"$apiId\"\n\nWrite-Highlight \"Setting the output variable 'Octopus.Action[$($stepName)].Output.ApiGatewayArn' to $sourceArn\"\nSet-OctopusVariable -name \"ApiGatewayArn\" -value \"$sourceArn\"\n"
```

Comment:
Line 88 in script: you use $ApiConnectionId not sure if you meant $ApiConnection ? I don't see where $ApiConnectionId is set.

Reply:
Ah, good catch, updated.